src/Time/History.hpp

```diff
@@ -0,0 +1,196 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <deque>
+#include <iterator>
+#include <pup.h>
+#include <tuple>
+#include <utility>
+
+#include "Parallel/PupStlCpp11.hpp"
+#include "Time/Time.hpp"
+
+namespace TimeSteppers {
+
+template <typename Vars, typename DerivVars>
+class HistoryIterator;
+
+/// \ingroup TimeSteppersGroup
+/// History data used by a TimeStepper.
+/// \tparam Vars type of variables being integrated
+/// \tparam DerivVars type of derivative variables
+template <typename Vars, typename DerivVars>
```

Comment:
can we default DerivVars?

Reply:
Turns out we can but doing so is not useful because this is almost always used in generic code that is already templated on Vars and DerivVars and passes them through to this class.  The only place where a default could be used is in one section of tests, and the required default would be `DerivVars=Vars` which is not the norm for evolutions.